src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -625,7 +625,10 @@ void program::pre_optimize_graph(bool is_internal) {
 
     apply_opt_pass<strided_slice_optimize>();
 
-    apply_opt_pass<handle_reshape>();
+    // When new shape inference is enabled, we need Reshape (unsqueeze) if Fully_connected inputs have different ranks
+    if (!options.get<build_option_type::allow_new_shape_infer>()->enabled()) {
+        apply_opt_pass<handle_reshape>();
```

Comment:
1. why do we need "reshape" for different ranks? reorder should be used for differnent ranks.. 
2. BTW, could you explain what kind of the case we need reshape, that should not be optimized by handle_reshape? 
3. Basically it is not expected to disable handle_reshape for allow_new_shape_infer